training_controller/prepare_training_logs.py

```diff
@@ -232,8 +229,26 @@ def upload_windows_tar_to_minio(self):
             "training-logs",
             os.path.basename(self.ES_DUMP_DIR_ZIPPED),
         )
+    def fetch_and_update_timestamps(self,es_instance):
+        oldest_log = es_instance.search(index="logs", body={"query": {"match_all": {}}, "_source": ["timestamp"],
```

Comment:
would this be a super heavy query to search and sort past 7 days' log?

Reply:
I think it will be fine. Anyways we are removing older logs automatically from Opendistro so I think it will be okay.

Reply:
is this the most efficient query to do this?
you should limit the search to maybe the 1 or 3 oldest logs so that there you aren't fetching a lot (or whatever the default #) of documents

Reply:
So, this is the full query that's being done 
```
        oldest_log = es_instance.search(index="logs", body={"query": {"match_all": {}}, "_source": ["timestamp"],
                           "sort": [{"timestamp": {"order": "asc"}}]}, size=1)```
where I set the limit on retrieval of logs to just 1 so it will be fetching the oldest log present within elasticsearch.

Reply:
so I am setting the size to 1 so it will always just fetch the oldest logs present within ES.